examples/platforms/nrf528xx/src/radio.c

```diff
@@ -971,3 +1076,34 @@ uint32_t nrf_802154_random_get(void)
 {
     return otRandomNonCryptoGetUint32();
 }
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+void otPlatRadioSetMacKey(otInstance *    aInstance,
+                          uint8_t         aKeyIdMode,
+                          uint8_t         aKeyId,
+                          const otMacKey *aPrevKey,
+                          const otMacKey *aCurrKey,
+                          const otMacKey *aNextKey)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aKeyIdMode);
+
+    assert(aPrevKey != NULL && aCurrKey != NULL && aNextKey != NULL);
+
+    if (nrf_802154_critical_section_enter())
```

Comment:
We should not use this function. I do see two possibilities, either:

1) Have a double buffering for keys, and make simple switch between them after keys are coppied.
2) Use `CRITICAL_REGION_ENTER();` and `CRITICAL_REGION_EXIT();` macros, which will handle also cases where SoftDevice is used. I believe the proper headers are already included.

Reply:
Thank you for reminding me. Updated.